File: 01-python/game.py
import random


NUMERO_MINIMO = 1
NUMERO_MAXIMO = 100


class JuegoAdivinaNumero:
    """Guarda el estado y contiene la lógica del juego."""

    def __init__(self):
        self.numero_secreto = 0
        self.intentos = 0
        self.reiniciar()

    def comprobar_numero(self, numero):
        if not isinstance(numero, int) or isinstance(numero, bool):
            raise ValueError("El intento debe ser un número entero.")

        if numero < NUMERO_MINIMO or numero > NUMERO_MAXIMO:
            raise ValueError("El número debe estar entre 1 y 100.")

        self.intentos += 1

        if numero < self.numero_secreto:
            return "Muy bajo"

        if numero > self.numero_secreto:
            return "Muy alto"

        return "¡Correcto!"

    def reiniciar(self):
        self.numero_secreto = random.randint(NUMERO_MINIMO, NUMERO_MAXIMO)
        self.intentos = 0

File: 09-c/game.c
#include "game.h"

#include <stdlib.h>
#include <time.h>

void iniciar_juego(Juego *juego)
{
    static bool semilla_inicializada = false;

    if (!semilla_inicializada) {
        srand((unsigned int)time(NULL));
        semilla_inicializada = true;
    }

    reiniciar_juego(juego);
}

void reiniciar_juego(Juego *juego)
{
    juego->numero_secreto = rand() % (NUMERO_MAXIMO - NUMERO_MINIMO + 1)
                           + NUMERO_MINIMO;
    juego->intentos = 0;
    juego->finalizado = false;
}

const char *comprobar_numero(Juego *juego, int numero)
{
    if (juego->finalizado || numero < NUMERO_MINIMO || numero > NUMERO_MAXIMO) {
        return NULL;
    }

    juego->intentos++;

    if (numero < juego->numero_secreto) {
        return "Muy bajo";
    }

    if (numero > juego->numero_secreto) {
        return "Muy alto";
    }

    juego->finalizado = true;
    return "¡Correcto!";
}
